src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drv.c

```diff
@@ -708,7 +708,11 @@ int xocl_refresh_subdevs(struct xocl_dev *xdev)
 		offset += resp->size;
 	} while (resp->rtncode == XOCL_MSG_SUBDEV_RTN_PARTIAL);
 
-	if (resp->rtncode == XOCL_MSG_SUBDEV_RTN_UNCHANGED &&
+	if (resp->rtncode == XOCL_MSG_SUBDEV_RTN_PENDINGPLP) {
+		(void) xocl_program_shell(xdev, true);
+		ret = -EAGAIN;
+		goto failed;
+	} else if (resp->rtncode == XOCL_MSG_SUBDEV_RTN_UNCHANGED &&
```

Comment:
I'd suggest you also explicitly check for XOCL_MSG_SUBDEV_RTN_COMPLETE return code so that next time xclmgmt return something new, old xocl can still gracefully error out.

Reply:
added the check